python/tests/test_client.py

```diff
@@ -5,7 +5,7 @@
 
 def test_multipart_regex():
 
-    client = Client(input_names=['partname'], output_names=['nobody_cares'])
+    client = Client()
```

Comment:
that breaks the test. you can remove the output names, but not the input

Reply:
Hmm. Maybe then I'll just leave the input names and output names arguments in there and if they're None then I'll try to infer them from the server

Reply:
I've implemented the above